services/db-backup/backup.sh

```
@@ -0,0 +1,78 @@
+
+# ── Upload ────────────────────────────────────────────────────────────────────
+aws s3 cp "${ENDPOINT_ARG[@]}" "$DUMP_FILE" "s3://${S3_BUCKET}/${S3_KEY}"
+rm -f "$DUMP_FILE"
+
+# ── Prune backups older than KEEP_DAYS ────────────────────────────────────────
+# Filename format: chess-YYYY-MM-DDTHH-MM-SSZ.sql.gz  (the T delimiter and the
+# hyphen-separated time are ISO-8601 with colons replaced so the filename is
+# cross-filesystem-safe). We reconstruct the real ISO timestamp and compare to
+# a cutoff epoch.
+CUTOFF_EPOCH=$(date -u -d "${KEEP_DAYS} days ago" +%s)
+echo "[backup] pruning backups older than ${KEEP_DAYS} days (< $(date -u -d @${CUTOFF_EPOCH} +%FT%TZ))"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)